src/bp/core/messaging/messaging-webhooks.ts

```diff
@@ -0,0 +1,130 @@
+import axios from 'axios'
+import { IO } from 'botpress/sdk'
+import { EventEngine } from 'core/events'
+import { TYPES } from 'core/types'
+import { inject, injectable, postConstruct } from 'inversify'
+import { AppLifecycle, AppLifecycleEvents } from 'lifecycle'
+import { ConversationService } from './conversation-service'
+import { MessageService } from './message-service'
+
+@injectable()
+export class MessagingWebhooks {
+  constructor(
+    @inject(TYPES.EventEngine) private eventEngine: EventEngine,
+    @inject(TYPES.ConversationService) private conversationService: ConversationService,
+    @inject(TYPES.MessageService) private messageService: MessageService
+  ) {}
+
+  @postConstruct()
+  async init() {
+    await AppLifecycle.waitFor(AppLifecycleEvents.CONFIGURATION_LOADED)
+
+    // TODO: register webhooks by http
+    const incomingWebhooks = [
+      {
+        id: 'mfkt837834her',
+        url: 'http://localhost:3020/hook',
+        // TODO: This could be a regex (but regex is kind of garbadge to write)
+        // This scope could be even further scoped for a bot: 'app.myBotId.userMessage'
+        scopes: ['app.userMessage']
+      }
+    ]
+    const outgoingWebhooks = [{ id: 'ef98484ufeb93', url: 'http://localhost:3020/hook', scopes: ['app.botMessage'] }]
+
+    // TODO: we need to include the api key when calling the webhooks
+    // TODO: why is the api key defined per workspace user??
+    // const http = axios.create({ headers: { 'x-bp-api-key': key } })
```

Comment:
I don't understand why we define api keys per workspace user.

I see two options : 
- We use an api key along with an appId. The appId can be related or not to a workspace user, but I don't want our api keys to be dependant on workspace users
- We define api keys per webhook (so when you send a request to make a new webhook, you get a new key back aswell. That's how smooch does it).